pyaerocom/web/helpers_evaluation_iface.py

```diff
@@ -467,6 +487,188 @@ def _init_ts_data():
+
+    min_month = data['time.month'].values.min()
+    max_month = data['time.month'].values.max()
+    months = range(min_month,max_month+1)
+    first_pass = True
+    for seas in seasons:
+    #for i,j in zip(months[b1::step], months[b2::step]):
+        rep_week_ds = xr.Dataset()
+        if resolution == 'seasonal':
+            mon_slice = data.where(data['time.season']==seas,drop=True)
+        elif resolution == 'yearly':
+            mon_slice = data
+        else:
+            raise NotImplementedError(f'Functionality currently not implemented for resolution {resolution}')
+        # mon_slice = data.where((data['time.month']>=i)&(data['time.month']<=j),drop=True)
+        month_stamp = f'seas'
+        
+        for day in range(7):
+            day_slice = mon_slice.where(mon_slice['time.dayofweek']==day,drop=True)
+            rep_day = day_slice.groupby('time.hour').mean(dim='time')
+            rep_day['hour'] = rep_day.hour/24+day+1
+            if day == 0:
+                rep_week = rep_day
+            else:
+                rep_week = xr.concat([rep_week,rep_day],dim='hour')
+                
+        rep_week=rep_week.rename({'hour':'dummy_time'})
+        month_stamps = np.zeros(rep_week.dummy_time.shape,dtype='<U5')
+        month_stamps[:] = month_stamp
+        rep_week_ds['rep_week']=rep_week
+        rep_week_ds['month_stamp'] = (('dummy_time'),month_stamps)
+        
+        if first_pass:
+            first_pass = False
+            rep_week_full_period = rep_week_ds
+        else:
+            rep_week_full_period = xr.concat([rep_week_full_period,rep_week_ds],dim='period')
+    return rep_week_full_period
+
+def _process_sites_weekly_ts(coldata,regions_how,region_ids,meta_glob):
+    #import xarray as xr
+    data = coldata.data
+    ts_objs = []
+    
+    if isinstance(coldata, ColocatedData):
+        _check_flatten_latlon_dims(coldata)
+        assert coldata.dims == ('data_source', 'time', 'station_name')
+
+    # rep_week_monthly = _create_diurnal_weekly_data_object(data, 'monthly')
+    # rep_week_seasonal = _create_diurnal_weekly_data_object(data, 'seasonal')
+
+    repw_res = {'seasonal':_create_diurnal_weekly_data_object(data, 'seasonal')['rep_week'],
+                'yearly':_create_diurnal_weekly_data_object(data, 'yearly')['rep_week'].expand_dims('period',axis=0),}
+
+    default_regs = get_all_default_regions(use_all_in_ini=False)
+
+
+    lats = repw_res['seasonal'].latitude.values.astype(np.float64)
+    lons = repw_res['seasonal'].longitude.values.astype(np.float64)
+    
+    if 'altitude' in repw_res['seasonal'].coords:
+        alts = repw_res['seasonal'].altitude.values.astype(np.float64)
+    else:
+        alts = [np.nan]*len(lats)
+    
+    if regions_how == 'country':
+        countries = repw_res['seasonal'].country.values
+    dc = 0
+    time = (np.arange(168)/24+1).round(4).tolist()
+    for i, stat_name in enumerate(repw_res['seasonal'].station_name.values):
```

Comment:
Consider moving the body of this for loop into a separate helper function?

Reply:
Done

Reply:
Good suggestion. I've done that as well.